### PR TITLE
Fix gcc compile issues

### PR DESCRIPTION
**Pull request by nvvishanthi**
Fixed various warnings

### CODE REVIEW

File: cudamapper/src/cudamapper_utils.cpp [hunk 1]
```diff
@@ -38,7 +38,7 @@ std::vector<gw_string_view_t> split_into_kmers(const gw_string_view_t& s, const
     const std::size_t kmer_count = s.length() - kmer_size + 1;
     std::vector<gw_string_view_t> kmers;
 
-    if (s.length() < kmer_size)
+    if (s.length() < static_cast<size_t>(kmer_size))
```
Comment (mimaric) [inline cudamapper/src/cudamapper_utils.cpp]: We actually have a functions which gets signed size of an array: https://github.com/clara-parabricks/GenomeWorks/blob/df8b1f38e48e2aacc13967cfa863f036c1dd37b4/common/base/include/claraparabricks/genomeworks/utils/signed_integer_utils.hpp#L31

File: cudamapper/src/overlapper.cpp [hunk 2]
```diff
@@ -306,7 +306,8 @@ void Overlapper::rescue_overlap_ends(std::vector<Overlap>& overlaps,
                                      const std::int32_t extension,
                                      const float required_similarity)
 {
-
+    (void)extension;
```
Comment (mimaric) [inline cudamapper/src/overlapper.cpp:309]: Eric should check why these values are not being used. I'll ping him directly as I cannot add him to this PR

File: cudamapper/src/utils.cpp [hunk 3]
```diff
@@ -161,8 +161,8 @@ static inline void encode_seq(bam1_t* const alignment, const std::string& seq)
     // encode bases as nibbles and write them into the alignment struct 2 bases at a time
     // 4 high bits are first base, 4 low bits are second base
     uint8_t* bam_seq_start = bam_get_seq(alignment);
-    int i                  = 0;
-    for (i; (i + 1) < len; i += 2)
+    uint32_t i             = 0;
```
Comment (mimaric) [inline cudamapper/src/utils.cpp]: Again, we tend to use signed values whenever possible (as suggested by ahehn-nv), so you can use https://github.com/clara-parabricks/GenomeWorks/blob/df8b1f38e48e2aacc13967cfa863f036c1dd37b4/common/base/include/claraparabricks/genomeworks/utils/signed_integer_utils.hpp#L31

**Review by mimaric — CHANGES_REQUESTED** [review]

File: cudamapper/src/overlapper.cpp (hunk 2 above)
Comment (edawson) [reply cudamapper/src/overlapper.cpp:309]: I can't seem to add comments to the PR, but these values should be used in the call to `details::overlapper::extend_overlap_by_sequence_similarity(overlap, query_view, target_view, 100, 0.9)`. I'm not sure why they are hard-coded as extension=100 and required_similarity=0.9 but it's quite possibly my mistake.

**Comment by nvvishanthi** [comment]
Closing this PR and moving to the PR in main genomeworks repo

**Pull request closed without merging**